File: restring/src/test/java/com/ice/restring/shadow/MyShadowAsyncTask.java
package com.ice.restring.shadow;

import android.os.AsyncTask;

import org.robolectric.annotation.Implements;
import org.robolectric.shadows.ShadowAsyncTask;

import java.util.concurrent.Executor;

@Implements(AsyncTask.class)
public class MyShadowAsyncTask<Params, Progress, Result> extends ShadowAsyncTask<Params, Progress, Result> {

    @Override
    public AsyncTask<Params, Progress, Result> executeOnExecutor(Executor executor, Params... params) {
        return super.execute(params);
    }
}

File: restring/src/test/java/dev/b3nedikt/restring/shadow/MyShadowAsyncTask.kt
package dev.b3nedikt.restring.shadow

import android.os.AsyncTask
import org.robolectric.annotation.Implements
import org.robolectric.shadows.ShadowLegacyAsyncTask
import java.util.concurrent.Executor

@Implements(AsyncTask::class)
class MyShadowAsyncTask<Params, Progress, Result> : ShadowLegacyAsyncTask<Params, Progress, Result>() {


    override fun executeOnExecutor(executor: Executor, vararg params: Params): AsyncTask<Params, Progress, Result> {
        return super.execute(*params)
    }

}
